src/cryptoKeyUtil.ts

```diff
@@ -0,0 +1,69 @@
+import { logger } from "@blockr/blockr-logger";
+import { ec as EC } from "elliptic";
+import { injectable } from "inversify";
+import { KeyPairException } from "./exceptions";
+import { SignatureException } from "./exceptions/signature.exception";
+
+@injectable()
+export class CryptoKeyUtil {
+    private ec: EC;
+
+    constructor() {
+        this.ec = new EC("secp256k1");
+    }
+
+    /**
+     * Generate a new (EC) crypto key pair.
+     * @returns {EC.KeyPair | IKeyPair} EC key pair.
+     * @throws {KeyPairException} when the generated key pair fails to verify.
+     */
+    public generateKeyPair(): EC.KeyPair {
+        logger.info("Generating key pair.");
+
+        const keyPair = this.ec.genKeyPair();
+
+        const validationResult = keyPair.validate();
+
+        if (validationResult.result) {
+            logger.info("Generated key pair successfully.");
+
+            return keyPair;
+        }
+
+        throw new KeyPairException("Keypair generation failed", validationResult.reason);
```

Comment:
The usage of "key pair / keypair" is inconsistent. Please stick to either "key pair" or "keypair" throughout the entire library, including the docs.

Reply:
Yes it should be "key pair".